src/generic-provider-views/index.js

```diff
@@ -253,6 +254,27 @@ module.exports = class View {
     })
   }
 
+  handleAuth () {
+    let urlId = Math.floor(Math.random() * 999999) + 1
+    let redirect = `${location.href}${location.search ? '&' : '?'}id=${urlId}`
+
+    const authState = btoa(JSON.stringify({ redirect }))
+    const link = `${this.plugin.opts.host}/connect/${this.Provider.authProvider}?state=${authState}`
+
+    let authWindow = window.open(link, '_blank')
+
+    let checkAuth = () => {
```

Comment:
I’d probably prefer to use `let` only if we need to set/reset later, `const` would seem more appropriate here.